tests/langchain/sample_code/openai_agent.py

```diff
@@ -19,22 +19,28 @@ class FakeOpenAI(ChatOpenAI, extra="allow"):
     # Therefore, we mock the OpenAI client in the model definition here.
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
+        self._reset()
 
+    def _stream(self, *args, **kwargs):
+        try:
+            chunk = next(self._responses)
+        except StopIteration:
+            self._reset()
+            chunk = next(self._responses)
+
+        yield ChatGenerationChunk(message=chunk)
```

Comment:
can we use `itertools.cycle`?

https://docs.python.org/3/library/itertools.html#itertools.cycle

Reply:
ooh this is exactly what I wanted